libraries/engage/product/components/Swatch/Swatch.jsx

```diff
@@ -0,0 +1,113 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import defaultsDeep from 'lodash/defaultsDeep';
+import { css } from 'glamor';
+import { PRODUCT_SWATCH } from '@shopgate/pwa-common-commerce/product/constants/Portals';
+import { SurroundPortals } from '../../../components';
+import { useConfig, useWidgetConfig } from '../../../core';
+import Element from './Element';
+
+const { colors = {} } = useConfig();
+
+const widgetDefaults = {
+  settings: {
+    selectionStyles: {
+      selected: {
+        borderColor: colors.accent,
+      },
+      unselected: {
+        borderColor: colors.shade4,
+      },
+    },
+    elementWidth: '1.5rem',
+    elementHeight: '1.5rem',
+  },
+  styles: {
+    swatch: {
+      marginBottom: '0.4rem',
+      display: 'flex',
+      flexDirection: 'row',
+      flexWrap: 'wrap',
+      justifyContent: 'flex-start',
+      alignItems: 'flex-start',
+      alignContent: 'flex-start',
+    },
+    element: {
+      flexGrow: 0,
+      flexShrink: 0,
+      flexBasis: 'auto',
+      alignSelf: 'auto',
+      marginRight: '0.3rem',
+      marginBottom: '0.3rem',
+      pointerEvents: 'none',
+      borderRadius: '50%',
+      backgroundSize: 'cover',
+      borderWidth: '1px',
+      borderStyle: 'solid',
+    },
+  },
+};
+
+const widgetId = '@shopgate/engage/product/Swatch';
+
+/**
+ * The swatch component.
+ * @param {Object} props The component props.
+ * @returns {JSX}
+ */
+const Swatch = ({ testId, swatch, widgetPath }) => {
+  const { settings = {}, styles = {} } = useWidgetConfig(widgetId, widgetPath);
+
+  // override default settings with configured widget settings
+  const widgetSettings = defaultsDeep(settings, widgetDefaults.settings);
+  const widgetStyles = defaultsDeep(styles, widgetDefaults.styles);
+
+  return (
+    <SurroundPortals portalName={PRODUCT_SWATCH} portalProps={{ swatch }}>
+      <ul data-test-id={testId} className={css(widgetStyles.swatch).toString()}>
```

Comment:
List does not support test id, to check if the swatch itself is rendered or not. This is fixable, but List.Item introduces own, concurrent styles and does not fulfill our needs in terms of display states because we need a four-way display and not just a boolean. 
I don't think List is supposed to be used that way.